app/frontend/src/components/designer/FormViewer.vue

```diff
@@ -1,117 +1,150 @@
-          <template #button-text-continue>
-            <span>Submit</span>
-          </template>
-        </BaseDialog>
+          <slot name="alert" v-bind:form="form" />
 
-        <Form
-          :form="formSchema"
-          :key="reRenderFormIo"
-          :submission="submission"
-          @submit="onSubmit"
-          @submitDone="onSubmitDone"
-          @submitButton="onSubmitButton"
-          @customEvent="onCustomEvent"
-          :options="viewerOptions"
-        />
-        <p v-if="version" class="text-right">Version: {{ version }}</p>
+          <BaseDialog
+            v-model="showSubmitConfirmDialog"
+            type="CONTINUE"
+            :enableCustomButton="canSaveDraft"
+            @close-dialog="showSubmitConfirmDialog = false"
+            @continue-dialog="continueSubmit"
+          >
+            <template #title>Please Confirm</template>
+            <template #text>Are you sure you wish to submit your form?</template>
+            <template #button-text-continue>
+              <span>Submit</span>
+            </template>
+          </BaseDialog>
+          <v-alert
+            v-if="isLoading && !bulkFile && submissionId == undefined"
+            class="mt-2 mb-2"
+            :value="isLoading"
+            :class="NOTIFICATIONS_TYPES.INFO.class"
+            :color="NOTIFICATIONS_TYPES.INFO.color"
+            :icon="NOTIFICATIONS_TYPES.INFO.icon"
+            transition="scale-transition"
+          >
+            <div color="info" icon="$info">
+              <v-progress-linear :indeterminate="true" color="blue-grey lighten-4" height="5"></v-progress-linear>
+              Please wait while the form is loading !!!
+            </div>
+          </v-alert>
+          <FormViewerMultiUpload
+            v-if="!isLoading && allowSubmitterToUploadFile && bulkFile"
+            :response="sbdMessage"
+            :formElement="formElement"
+            :form="form"
+            :formSchema="formSchema"
+            :json_csv="json_csv"
+            @save-bulk-data="saveBulkData"
+            @reset-message="resetMessage"
+            @set-error="setError"
+            :formFields="formFields"
+          />
+          <Form
+            class="mt-4"
+            v-if="!bulkFile"
+            ref="chefForm"
+            :form="formSchema"
+            :key="reRenderFormIo"
+            :submission="submission"
+            :options="viewerOptions"
+            @submit="onSubmit"
+            @submitDone="onSubmitDone"
+            @submitButton="onSubmitButton"
+            @customEvent="onCustomEvent"
+            @change="formChange"
+            @render="onFormRender"
+          />
+          <p v-if="version" class="text-right">Version: {{ version }}</p>
+        </div>
       </div>
-    </div>
-  </v-skeleton-loader>
+      <YesOrNoDialog :doYouWantToSaveTheDraft="doYouWantToSaveTheDraft" @save-draft="saveDraftFromModal" @close-bulk-yes-or-no="closeBulkYesOrNo" />
```

Comment:
Because there are  some change in the language  like if you want to save draft yes or no but the BaseDialog have [continue and cancel ] and there is no way to custom the message.